src/runtime_src/core/pcie/driver/linux/xocl/userpf/xocl_ioctl.c

```diff
@@ -292,6 +292,37 @@ static bool xclbin_downloaded(struct xocl_dev *xdev, xuid_t *xclbin_id)
 	return ret;
 }
 
+static int xocl_preserve_memcmp(struct mem_topology *new_topo, struct mem_topology *mem_topo, size_t size)
+{
+	int ret = -1;
+	size_t i, j = 0;
+
+	if (mem_topo->m_count != new_topo->m_count)
+		return ret;
+
+	for (i = 0; i < mem_topo->m_count; ++i) {
+		if (IS_HOST_MEM(mem_topo->m_mem_data[i].m_tag))
+			continue;
+		for (j = 0; j < new_topo->m_count; ++j) {
+			if (IS_HOST_MEM(new_topo->m_mem_data[j].m_tag))
```

Comment:
shouldn't you compare the m_tag here? we don't want to assume the order of mem entries are the same b/w these two xclbins, right?

Reply:
talked to Chien-Wei offline. the code here is not efficient, but functionality is OK. approving it now.